### PR TITLE
Fixed mutiselect for uninitialied years

### PR DESCRIPTION
**Pull request by roylin2000**
# Description

Fixed bug where a selected date that is in an uninitialized year would cause an error. Now, LogMulitpleDayPeriod() will first initialize an empty year in those cases.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (if required)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: tpp-app/src/services/LogSymptomsService.js [hunk 1]
```diff
@@ -84,41 +89,52 @@ export const LogMultipleDayPeriod = async (dates) => new Promise(async (resolve,
...
-            // a bit unneccessary since you can't log symptoms for the future.
-            if(calendarData[curYear + 1]){
-                AsyncStorage.setItem((curYear + 1).toString(), JSON.stringify(calendarData[curYear + 1]))
-                .then(() => resolve())
-                .catch((e) => {
-                    reject(`Unable to mergeItem and post symptoms for multiselect.`);
-                    console.log(JSON.stringify(e));
-                });
-            }
+            // if(calendarData[curYear]){
```
Comment (annie951977) [inline tpp-app/src/services/LogSymptomsService.js]: Why not just delete this?

Comment (roylin2000) [reply tpp-app/src/services/LogSymptomsService.js]: True I totally forgot about it when I fixed it lol

Comment (roylin2000) [reply tpp-app/src/services/LogSymptomsService.js]: removed!

**Review by annie951977 — APPROVED** [review]
pogchamp

**Review by faye-tan — APPROVED** [review]

**Pull request merged**